src/util.js

```diff
@@ -65,4 +97,134 @@ export default class Utils {
+
+        let type;
+        let buf;
+        if (value < min_32 || value > max_32) {
+            type = thrift.annotationType.I64;
+            buf = Utils.encodeInt64(value, 0);
+        } else if (value < min_16 || value > max_16) {
+            type = thrift.annotationType.I32;
+            buf = new Buffer(4);
+            buf.writeInt32BE(value, 0);
+        } else {
+            type = thrift.annotationType.I16;
+            buf = new Buffer(2);
+            buf.writeInt16BE(value, 0);
+        }
+
+        return this.createBinaryAnnotation(key, buf, type);
+    }
+
+    /**
+     * @param {number} timestamp - number to be stored in this annotation
+     * @param {string} value - the string value stored in this annotation
+     * @param {Endpoint} host - the Endpoint stored in this annotation
+     * @return {Annotation} - an annotation representing the parameters received.
+     **/
+    static createAnnotation(timestamp: number, value: string , host?: Endpoint): Annotation {
+        return {
+            timestamp: timestamp,
+            value: value,
+            host: host
+        };
+    }
+
+    /**
+     * A function which creates logged data which is backed by a zipkin annotation.
+     *
+     * @param {number} timestamp - number to be stored in this log
+     * @param {string} name - the name stored in this log 
+     * @param {Endpoint} host - the Endpoint stored in this log
+     * @return {Annotation} - an annotation representing the parameters received.
+     **/
+    static createLogData(timestamp: number, name: string, host?: Endpoint): Annotation {
+        return this.createAnnotation(timestamp, name, host);
+    }
+
+    /**
+     * A function which creates boolean tags backed by zipkin binary annotations
+     *
+     * @param {string} key - number to be stored in this log
+     * @param {boolean} value - the name stored in this log 
+     * @return {BinaryAnnotation} - a string tag represented by a binary annotation.
+     **/
+    static createBooleanTag(key: string, value: boolean): BinaryAnnotation {
+        let booleanValue = '0x0';
+        if (value) {
+            booleanValue = '0x1';
+        }
+
+        return this.createBinaryAnnotation(key, booleanValue, thrift.annotationType.BOOL);
+    }
+
+    /**
+     * A function which creates string tags backed by zipkin binary annotations
+     *
+     * @param {string} key - number to be stored in this log.
+     * @param {boolean} value - the name stored in this log 
+     * @return {BinaryAnnotation} - a string tag represented by a binary annotation.
+     **/
+    static createStringTag(key: string, value: string): BinaryAnnotation {
+        return this.createBinaryAnnotation(key, value, thrift.annotationType.STRING);
+    }
+
+    /**
+     * Returns the timestamp in microseconds.
+     * @return {number} - The microseconds since the epoch.
+     **/
+    static getTimestampMicros(): number {
+        return Date.now() * 1000;
+    }
```

Comment:
please add a todo to investigate `process.hrtime()`, per http://stackoverflow.com/questions/11725691/how-to-get-a-microtime-in-node-js